packages/ckeditor5-code-block/src/codeblockediting.ts

```diff
@@ -277,6 +280,43 @@ export default class CodeBlockEditing extends Plugin {
 			data.preventDefault();
 			evt.stop();
 		}, { context: 'pre' } );
+
+		this._initAriaAnnouncements( );
+	}
+
+	/**
+	 * Observe when user enters or leaves code block and set proper aria value in global live announcer.
+	 * This allows screen readers to indicate when the user has entered and left the specified code block.
+	 *
+	 * @internal
+	 */
+	private _initAriaAnnouncements( ) {
+		const { model, ui, t } = this.editor;
+		const languageDefs = getNormalizedAndLocalizedLanguageDefinitions( this.editor );
+
+		let lastFocusedCodeBlock: Element | null = null;
+
+		model.document.selection.on<SelectionChangeRangeEvent>( 'change:range', () => {
```

Comment:
This works fine as long as the user moves between a code block and a non-code block context. If they move between two code blocks (create two code blocks, put the caret in the last line of the first one, then press arrow down), only the "Entering..." part is read out, while it should be "Leaving..." followed by "Entering...".

It's the same story as with the comment markers.

Reply:
Fixed